simulations/vip-192/vip-192/simulations.ts

```diff
@@ -0,0 +1,191 @@
+          type: "address",
+        },
+        {
+          indexed: false,
+          internalType: "address",
+          name: "newImplementation",
+          type: "address",
+        },
+      ],
+      name: "NewImplementation",
+      type: "event",
+    },
+  ];
+  testVip("VIP-192 Core  VToken Upgrade of AIA", vip192(), {
+    callbackAfterExecution: async txResponse => {
+      await expectEvents(
+        txResponse,
+        [VTOKEN_ABI, ProxyAdminInterface],
+        ["NewImplementation", "NewProtocolShareReserve", "NewReduceReservesBlockDelta", "NewAccessControlManager"],
+        [22, 22, 22, 22],
+      );
+    },
+  });
+});
+
+forking(32906245, () => {
+  describe("Post VIP simulations", async () => {
+    before(async () => {
+      await pretendExecutingVip(vip192());
+      [user] = await ethers.getSigners();
+    });
+
+    for (const market of CORE_MARKETS) {
+      it(`Save post VIP storage snapshot of ${market.name}`, async () => {
+        user = await initMainnetUser(market.holder, ethers.utils.parseEther("5"));
+
+        vToken = new ethers.Contract(market.address, VTOKEN_ABI, provider);
+        const comptroller = new ethers.Contract(await vToken.comptroller(), COMPTROLLER_ABI, provider);
+        underlying = new ethers.Contract(await vToken.underlying(), MOCK_TOKEN_ABI, provider);
+
+        await comptroller.connect(impersonatedTimelock)._setMarketBorrowCaps([market.address], [parseUnits("2", 48)]);
+        await comptroller.connect(impersonatedTimelock)._setMarketSupplyCaps([market.address], [parseUnits("2", 48)]);
+        await comptroller.connect(impersonatedTimelock)._setCollateralFactor(market.address, parseUnits("0.95", 18));
+
+        // Some actions are paused
+        if (market.name != "vBETH" && market.name != "vSXP") {
+          await performVTokenBasicActions(
+            market.address,
+            user,
+            mintAmount,
+            borrowAmount,
+            repayAmount,
+            redeemAmount,
+            vToken,
+            underlying,
+            false,
+          );
+        }
+        const state = await fetchVTokenStorageCore(vToken, user.address);
+
+        delete state.totalReserves;
+        delete state.pendingOwner;
+        delete state.owner;
+
+        expect(await vToken.implementation()).equals(NEW_VBEP20_DELEGATE_IMPL);
+        expect(await vToken.protocolShareReserve()).equals(PROTOCOL_SHARE_RESERVE);
+        expect(await vToken.accessControlManager()).equals(ACCESS_CONTROL_MANAGER);
+        expect(await vToken.admin()).equals(NORMAL_TIMELOCK);
+
+        postVipStorage.push(state);
+      });
+    }
+
+    it("Should match pre and post storage", async () => {
+      for (let i = 0; i < preVipStorage.length; i++) {
+        expect(preVipStorage[i]).to.deep.equal(postVipStorage[i]);
+      }
+    });
+  });
+});
+
+// In very first operation after upgrade the reserves will be reduced (delta > lastReduceReservesBlockNumber(0)).
+forking(32906245, () => {
```

Comment:
I would add a simulation similar to this into `simulations/vip-192/vip-192-testnet/simulations.ts`